test/transactsql.test.ts

```diff
@@ -118,27 +118,37 @@ describe('TransactSqlFormatter', () => {
     `);
   });
 
-  it('formats SET NOCOUNT ON; GO CREATE OR ALTER TABLE', () => {
-    const result = format('SET NOCOUNT ON; GO CREATE OR ALTER TABLE t');
+  it('formats SET NOCOUNT ON; GO CREATE OR ALTER PROCEDURE', () => {
+    const result = format('SET NOCOUNT ON; GO CREATE OR ALTER PROCEDURE p');
     expect(result).toBe(dedent`
     SET
     NOCOUNT ON;
 
     GO
-    CREATE OR ALTER TABLE
-      t
+    CREATE OR ALTER PROCEDURE
+      p
     `);
```

Comment:
Probably best to concentrate this test to just the `GO` statement.

The preceding `SET` statement isn't really relevant here. All the `;`-separated statements are formatted in isolation anyway, so it can't really have any effect to the test outcome.

In general I haven't written tests for all the combinations of `OR ALTER`, `IF EXISTS` etc. Just too much effort for too little benefit. There are some generic tests that ensure the expansion of the `{blah | blah} [something]` syntax works properly.

Therefor in here I would just concentrate on testing `GO` with some simple statement that could follow it. (I personally don't really know that much about TSQL, and especially about the GO-statement).

Reply:
Fair enough, I have removed the `SET` statement from the test.

The problem I noticed with the `GO` statement was when I was defining the stored procedure with `CREATE OR ALTER` as mentioned in #606. Would this be acceptable?

Reply:
Ah yeah... I misunderstood things a little.

The actual problem you had wasn't really with the `GO` statement. But I think it looked like a problem with formatting of `GO` because the formatter probably formatted it all on one long line.

The thing is that the formatter doesn't actually detect the `GO` statement specifically. It only detects the `CREATE TABLE` or `CREATE PROCEDURE` and other such statements and adds a newline before them (not a newline after `GO`). It just treats `GO` as some unknown statement it doesn't understand and prints it without line breaks. And similarly it treated the whole `GO CREATE OR ALTER PROCEDURE p`.

This behavior is unfortunate, but on the other hand I'm reluctant to add a rule for formatting the `GO` keyword, as it's such a short one it could also easily occur in various other context of the SQL language, and the formatter is pretty poor at understanding the context.